android-buddy-plugin/src/main/java/com/likethesalad/android/buddy/utils/ConcatIterator.kt

```diff
@@ -1,8 +1,13 @@
 package com.likethesalad.android.buddy.utils
 
-class ConcatIterator<T>(val iterators: MutableList<out Iterator<T>>) : MutableIterator<T> {
+open class ConcatIterator<T>(val iterators: MutableList<out Iterator<T>>) : MutableIterator<T> {
 
     override fun hasNext(): Boolean {
+        while (iterators.isNotEmpty() && !iterators.first().hasNext()) {
```

Comment:
I'm a bit confused by this part with two things, the first condition that checks if the list isn't empty seems that wouldn't be needed if this `while` would be placed below the `if` on the line 11.

And the second thing is that, the removal of the empty iterator is done within the `next` function below, so this part seems to be duplicating that logic, do we need to modify the list in here or can we wait until the `next` function gets called instead?

Reply:
Oh! I see, this seems like a tricky bug to find, thanks for sorting it.